src/main/java/com/gluonhq/substrate/util/FileOps.java

```diff
@@ -675,27 +675,64 @@ public static void downloadAndUnzip(String sourceUrl, Path destPath, String file
     }
 
     /**
-     * Shorten the Java classpath with a pathing jar
+     * Shorten the Java classpath with a pathing jar. This works by creating a temporary
+     * empty jar file where the full classpath is defined in its Class-Path entry in
+     * the manifest. All files on the classpath will be copied to the same temporary
+     * folder, while all directories will be resolved relatively against that temporary
+     * folder. The Class-Path entry will ultimately contain all classpath elements as a
+     * reference that is relative to the pathing jar.
+     *
      * @param classpath A string with the classpath of files that will be added to the
      *                 pathing jar Class-Path attribute
      * @return a String with the path to the created pathing jar
      * @throws IOException
      */
-    public static String createPathingJar(String classpath) throws IOException {
+    public static String createPathingJar(Path tmpPath, String classpath) throws IOException {
         Objects.requireNonNull(classpath);
+
+        Files.createDirectories(tmpPath);
```

Comment:
I don't think it's necessary. For one, the actual items that are considered for the Class-Path manifest entry are taken from the current classpath and not from the list of files that already existed in this folder. And second, it's a temporary folder, so that means it should be short-lived anyway.